ytmusicapi/mixins/browsing.py

```diff
@@ -711,3 +710,41 @@ def get_signatureTimestamp(self, url: str = None) -> int:
             raise Exception("Unable to identify the signatureTimestamp.")
 
         return int(match.group(1))
+
+    def get_tasteprofiles(self) -> Dict:
+        """
+        Fetches all artists from taste profile (music.youtube.com/tasteprofile)
+
+        :return: Dictionary with artist and their selection & impression value
+
+        Example::
+        {
+            "Drake": {
+                "selectionValue": "tastebuilder_selection=/m/05mt_q",
+                "impressionValue": "tastebuilder_impression=/m/05mt_q"
+            }
+        }
+        """
+        response = self._send_request('browse', {'browseId': "FEmusic_tastebuilder"})
+        return parse_tasteprofiles(response)
+
+    def set_tasteprofile(self, *artists: str) -> None:
+        """
+        Favorites artists to see more recommendations from the artist
+        """
+
+        taste_profiles = self.get_tasteprofiles()
+        formData = {
+            "impressionValues":
+            [taste_profiles[profile]["impressionValue"] for profile in taste_profiles],
+            "selectedValues": []
+        }
+
+        for artist in artists:
+            if artist not in taste_profiles:
+                raise Exception("The artist, {}, was not present in taste!".format(artist))
+            formData["selectedValues"].append(taste_profiles[artist]["selectionValue"])
+
+        body = {'browseId': "FEmusic_home", "formData": formData}
+
+        self._send_request('browse', body)
```

Comment:
The function should return the YouTube Music response or error, not `None`.

Reply:
The return response does not contain any useful information that a user can use. Error is already handled inside of `self._send_request`. Though, if something must be returned, then an error would fit best.  A try/except block on the `self._send_request` would suffice.

Reply:
I checked the response myself and you are right, it contains merely the redirect to the home page. I don't think we need a try except around _send_request, as it's only throwing HTTP-type exceptions, which are needed. Consider this point resolved